enterprise/app/trends/heatmap.tsx

```diff
@@ -0,0 +1,422 @@
+import React from "react";
+import moment from "moment";
+
+import { stats } from "../../../proto/stats_ts_proto";
+import { ScaleBand, scaleBand } from "d3-scale";
+import { withResizeDetector } from "react-resize-detector";
+import { pinBottomLeftOffsetFromMouse, MouseCoords, Tooltip } from "../../../app/components/tooltip/tooltip";
+
+interface HeatmapProps {
+  heatmapData: stats.GetStatHeatmapResponse;
+  width: number;
+  height: number;
+  valueFormatter: (value: number) => string;
+  metricBucketName: string;
+  metricBucketFormatter: (value: number) => string;
+  selectionCallback?: (s?: HeatmapSelection) => void;
+}
+
+interface State {
+  selectionToRender?: [TooltipData, TooltipData];
```

Comment:
at first glance it's unclear why this is a pair of `TooltipData` rather than just one, mind adding a comment?

Reply:
Sure yes, and I changed the name as well because tooltipdata stinks and is dated.